File: pingyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbBrand;
import com.pinyougou.entity.PageResult;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有品牌数据
     *
     * @return
     */
    public List<TbBrand> findAll();

    /**
     * f分页查询
     *
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(int pageNum, int pageSize);

    /**
     * 添加品牌
     *
     * @param tbBrand
     */
    public void add(TbBrand tbBrand);

    /**
     * 根据id查询
     *
     * @param id
     * @return
     */
    public TbBrand findOne(long id);

    /**
     * 修改品牌
     *
     * @param tbBrand
     */
    public void update(TbBrand tbBrand);

    /**
     * 删除
     *
     * @param ids
     */
    public void delete(long[] ids);

    /**
     * 模糊查询
     *
     * @param tbBrand
     * @param pageNum
     * @param pageSize
     * @return
     */
    public PageResult findPage(TbBrand tbBrand, int pageNum, int pageSize);

    public List<Map> selectOptionList();
}

File: pingyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务层
app.service("brandService",function ($http) {

    //读取列表数据绑定表单
    this.findAll=function () {
        return $http.get("../brand/findAll.do");
    }

    //分页查询
    this.findPage=function (page,rows) {
        return $http.get("../brand/findPage.do?page="+page+"&rows="+rows);
    }
    //根据id查询
    this.findOne=function (id) {
        return $http.get("../brand/findOne.do?id="+id);
    }

    //增加
    this.add=function (entity) {
        return $http.post("../brand/add.do",entity)
    }
    //修改
    this.update=function (entity) {
        return $http.post("../brand/update.do",entity)
    }
    //删除
    this.dele=function (ids) {
        return $http.get("../brand/delete.do?ids="+ids);
    }

    //搜索
    this.search=function (page,rows,searchEntiry) {
        return $http.post("../brand/serch.do?page="+page+"&rows="+rows,searchEntiry);
    }

    //下拉列表
    this.selectOptionList=function () {
        return $http.get("../brand/selectOptionList.do");
    }
});